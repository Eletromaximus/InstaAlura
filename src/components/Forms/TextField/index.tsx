// eslint-disable-next-line no-use-before-define
import React from 'react'
import styled from 'styled-components'
import Text from '../../foundation/Text'
interface IInput {
	placeholder: string;
	name: string;
	type?: string;
	value: string;
	children?: string;
	isTouched?: boolean;
	error?: string;
	onBlur?: (event: React.ChangeEvent<HTMLInputElement>) => void;
	onChange: ((event: React.ChangeEvent<HTMLInputElement>) => void) | undefined;
}
const InputWrapper = styled.div`
	margin-bottom: 17px;
`

const Input = styled.input`
	width: 100%;
	border: 1px solid ${({ theme }) => theme.colors.tertiary.light.colors};
	padding: 12px 16px;
	outline: 0;
	border-radius: ${({ theme }) => theme.borderRadius};
`
export default function TextField ({
  isTouched, placeholder,
  error,
  name,
  value,
  ...props
}: IInput) {
  const hasError = Boolean(error)
  const isFieldInvalid = hasError && isTouched
  return (
		<InputWrapper>
			<Input
				placeholder={placeholder}
				name={name}
				value={value}
				{...props}
			/>

			{isFieldInvalid &&
				<Text
					variant='smallestException'
					color='error.main'
					// role='alert'
				>
					{error}
				</Text>
			}

		</InputWrapper>
  )
}

TextField.defaultProps = {
  error: '',
  isTouched: false
}
