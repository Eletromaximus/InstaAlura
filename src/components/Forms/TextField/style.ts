import styled, { css } from 'styled-components'
import React from 'react'

export interface IInput {
	placeholder: string;
	name: string;
	type?: string;
	value: string;
	children?: string;
	isTouched?: boolean;
	isFieldInvalid?: boolean;
	error?: string;
	onBlur?: (event: React.ChangeEvent<HTMLInputElement>) => void;
	onChange: ((event: React.ChangeEvent<HTMLInputElement>) => void) | undefined;
}

export const InputWrapper = styled.div`
margin-bottom: 17px;
`

export const Input = styled.input<IInput>`
width: 100%;
border: 1px solid ${({ theme }) => theme.colors.tertiary.light.colors};
padding: 12px 16px;
outline: 0;
border-radius: ${({ theme }) => theme.borderRadius};

${({ theme, isFieldInvalid }) => isFieldInvalid && css`
  border-color: ${theme.colors.error.main.color};
  & + span {
    color: ${theme.colors.error.main.color};
    font-size: 11px;
  }
`}
`
