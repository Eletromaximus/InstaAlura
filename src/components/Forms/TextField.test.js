import React from 'react'
import { render, screen } from '../../infra/tests/testUtils'
import TextField from './TextField'

describe('<TextField />', () => {
  test('renders component', () => {
    render(
        <TextField
        placeholder='Nome'
        value='Mario'
        onChange={() => {}}
        name='name'
        />
    )

    screen.debug()

    const textField = screen.getAllByPlaceholderText(/nome/i)

    console.log(textField)

    expect(textField).toMatchSnapshot()
  })
})
