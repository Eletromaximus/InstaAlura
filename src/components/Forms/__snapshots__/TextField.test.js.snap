// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> renders component 1`] = `
Array [
  .c0 {
  width: 100%;
  border: 1px solid;
  padding: 12px 16px;
  outline: 0;
  border-radius: 8px;
}

<input
    class="c0"
    name="name"
    placeholder="Nome"
    value="Mario"
  />,
]
`;
